content/en/blog/_posts/2021-03-09-Defining-NetworkPolicy-Conformance-For-CNIs

```diff
@@ -0,0 +1,373 @@
+---
+layout: blog
+title: "Defining NetworkPolicy Conformance for CNI providers"
+date: 2021-03-21
+slug: defining-networkpolicy-conformance-cni-providers
+---
+
+Authors: Matt Fennwick, Jay Vyas, Ricardo Katz, Amim Knabben, Douglas Landgraf
+
+A brief conversation around "node local" network policies in April of 2020 inspired the creation of a NetworkPolicy subproject from sig-network.
+
+In this post we'll discuss:
+
+- Why we created a Subproject for Network Policies 
+- How we changed the Kubernetes e2e framework to `visualize` NetworkPolicy implementation of your CNI provider
+- The initial results of our comprehensive NetworkPolicy Conformance validator, `Cyclonus`, built around these principles
+- Improvements we've made to the NetworkPolicy user experience
+
+## Why we created a subproject for NetworkPolicies
+
+In April of 2020 it was becoming clear that many CNIs were emerging, and many vendors implement these CNIs in different ways.  Users were beggining to express a little bit of confusion around how to implement policies for different scenarios, and asking for new features.
+It was clear that we needed to begin unifying the way we think about network policies in Kubernetes, if we wanted to avoid becoming an "ancient" API which was going to be overrun by implementation divergences. 
+
+For example:  
+- Calico as a CNI provider can be run using IPIP or VXLAN mode.  CNI's such as Antrea and Cilium offer similar configuration divergences as well.
+- Some CNIs rely on IPTables for NetworkPolicies (i.e. calico), whereas others use a completely different technology stack (for example, the antrea project uses Open vSwitch rules).
```

Comment:
```suggestion
- Some CNI plugins rely on iptables for NetworkPolicies (for example: Calico), whereas others use a completely different technology stack (for example, the Antrea project uses Open vSwitch rules).
```

Does Calico _rely_ on iptables? https://www.projectcalico.org/hands-on-with-calicos-ebpf-service-handling/ suggests it does not.

Reply:
I believe you can switch for both. Addressed the comment.